src/event-engine/presence/states/heartbeating.ts

```diff
@@ -0,0 +1,68 @@
+import { State } from '../../core/state';
+import { Events, disconnect, heartbeatFailure, heartbeatSuccess, joined, left, leftAll } from '../events';
+import { Effects, emitStatus, heartbeat, leave } from '../effects';
+import { HeartbeatCooldownState } from './heartbeat_cooldown';
+import { HearbeatReconnectingState } from './heartbeat_reconnecting';
+import { HeartbeatStoppedState } from './heartbeat_stopped';
+import { HeartbeatInactiveState } from './heartbeat_inactive';
+
+export type HeartbeatingStateContext = {
+  channels: string[];
+  groups: string[];
+};
+
+export const HeartbeatingState = new State<HeartbeatingStateContext, Events, Effects>('HEARTBEATING');
+
+HeartbeatingState.onEnter((context) => heartbeat(context.channels, context.groups));
+
+HeartbeatingState.on(heartbeatSuccess.type, (context, event) => {
+  return HeartbeatCooldownState.with(
+    {
+      channels: context.channels,
+      groups: context.groups,
+    },
+    [emitStatus(event.payload)],
```

Comment:
Do we have any status here right now? Or is it for backward compatibility?

Reply:
removed as per new specs ✔️